lib/metrics.py

```diff
@@ -348,3 +348,41 @@ def _get_db_conn(self):
                 psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT
             )
         return self.db
+
+
+class FreeAppsMetricsEmitterThread(MetricsEmitterThread):
+    def run(self):
+        logger.debug(
+            "Starting free apps metrics emitter with interval %d"
+            % self.interval
+        )
+        while True:
+            stats = {}
+            try:
+                try:
+                    stats = self._inject_m2ee_stats(stats)
+                except Exception:
+                    logger.debug("Unable to get metrics from runtime")
+                # Only push session metrics for free apps
+                self.emit(stats["mendix_runtime"]["sessions"])
+            except psycopg2.OperationalError as up:
+                logger.exception("METRICS: error while gathering metrics")
```

Comment:
We could also log the stacktrace here if there is no specific reason not to. Also in L379.
```suggestion
                logger.exception("METRICS: error while gathering metrics"., exc_info=True)
```
@cwesdorp would that cause any issues?

Reply:
Never mind, `logger.exception` already takes car of this: https://docs.python.org/3/library/logging.html#logging.Logger.exception